### DISCUSSION TITLE
Short living link - speed issues

### DISCUSSION THREAD

**Discussion by ghost** [minio/minio#17714] [Q&A] [answered]
Hello,

we developed an app similar to youtube, where for every resource (slide, tumbnail, video, etc.) you generate short living link. Now the issue is that this approach for 100 short living links takes like 0.1s per request. We use https with http it worked faster 0.01. How we could optimize that this will work faster or which another approach to use? Wich details would you need, to help me optimize more ?

**Comment by klauspost** [comment, marked as answer]
You provide no real information for us to help you. You can use [warp](https://github.com/minio/warp#get) to benchmark at different concurrency levels.  With `--analyze.v` it will print request time-to-first byte. The objects being "short-lived" shouldn't really matter.

That said there isn't too much we can help you without a [subscription](https://min.io/pricing), so consider that for a more detailed analysis. If you are doing a PoC, feel free to write `hello@min.io` and we can help you get set up.

**Comment by ghost** [comment]
@klauspost In what way you could help with PoC ?

**Reply by klauspost** [reply]
It really depends on your needs. We typically help with HW selection, so you don't overspend and still reach your performance targets. If you have specific performance targets (throughput, latency, concurrent operations) we can help optimize for that.

Of course we can also help with any questions with regards to configuration, S3 API/SDK or anything else related to that that may come up. Finally when you are up and running continuous monitoring of your system and dealing with anything that may come up.

**Comment by MatejKovacic** [comment]
I would test installation that uses HTTP to minio, but before the minio installation would put HTTPS proxy, so everything to the outside would be HTTPS encrypted, but internally (on localhost) it would work without encryption.

**Comment by ghost** [comment]
I tried this approach yesterday, but the issue was when I signed url `minio:9000`, than I couldnt use it for outer hostname. The question is if its possible now to somhow sign hostname/url and than replace with other.

So when I send signing url I would send directly to docker machine: minio:9000, but when backend return a response to frontend should be minio.domain.com/... Would that be possible ? @klauspost

**Comment by MatejKovacic** [comment]
I guess `minio` has entry in local `/etc/hosts` or local (internal) DNS, so if you call let's say `curl http://minio:9000` from the **local machine**, it will be accessible?

In that case I would suggest using Nginx reverse proxy like:
```
 server {
        server_name minio.example.com;
        location / {
            proxy_pass       http://hostname1:port1;
        }
 }
 server {
        server_name sometningelse.example.com;
        location / {
            proxy_pass       http://hostname2:port2;
        }
 }

```
In that case you can run `curl https://minio.example.com` from outside machine, and you should get the data through HTTPS proxy.

However, I guess, you should also set some `proxy_set_header`, maybe this would be sufficient (more or less guessing):

```
server {
    listen 443 default_server;

    root /var/www/html;
    index index.html index.htm;
    server_name _;

    ssl on;
    ssl_certificate ...
    ssl_certificate_key ...
    ...

    location / {
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $http_host;
        proxy_set_header X-Forwarded-Proto https;
        proxy_redirect off;
        proxy_pass http://192.168.100.51;
        proxy_http_version 1.1;
    }
}

```
